src/utils/hooks/use-mount-transition.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+
+type TransitionTrigger = 'in' | 'out';
+
+interface UseMountTransitionResult {
+  shouldRender: boolean;
+  transitionTrigger: TransitionTrigger;
+}
+// MEMO: Inspired by: https://stackoverflow.com/a/54114180
+
+const useMountTransition = (isMounted: boolean, transitionDuration: number): UseMountTransitionResult => {
+  const [shouldRender, setShouldRender] = React.useState(false);
+  const [transitionTrigger, setTransitionTrigger] = React.useState<'in' | 'out'>('in');
+
+  React.useEffect(() => {
+    let unmountDelayTimeout: NodeJS.Timeout;
+    let mountTransitionTimeout: NodeJS.Timeout;
+    if (isMounted) {
```

Comment:
Would be nice to have a bit of whitespace here :) I would tend to put a line break between the variables, the if/else statement, and the return.